src/main/java/org/candlepin/controller/EntitlementCertificateGenerator.java

```diff
@@ -328,6 +333,122 @@ public void regenerateCertificatesOf(Consumer consumer, boolean lazy) {
         this.regenerateCertificatesOf(consumer.getEntitlements(), lazy);
     }
 
+    @Transactional
+    public void regenerateCertificatesOf(Consumer consumer, List<String> updatedEnvironments,
```

Comment:
Instead of adding yet another overloaded `regenerateCertificatesOf` method in this class (there are 7 already), can we break all this functionality down a bit? I don't think `EntitlementCertificateGenerator.java` needs to know anything about filtering entitlements; its sole responsibility should be generating them.
I would suggest moving the methods `filterEntitlements` and `getContentUUIDsFromEntitlement` into their own class called something like `EntitlementEnvironmentFilter`, which will be used at the ConsumerResource.java level; and once the list of filtered is returned, we pass them into one of the existing `EntitlementCertificateGenerator` methods

Reply:
This has the added benefit of making the filtering functionality much more unit-testable